R/expect-generic.R

```diff
@@ -56,35 +54,77 @@
 #' )
 #' )
 #'
-#' # Check petal dimensions are positive
-#' expect_allany(
-#'   vars = where(is.numeric),
-#'   func = chk_range,
-#'   args = list(min = 0, max = Inf),
-#'   allany = chk_filter_all,
+#' # Check that all variables are numeric:
+#' try(expect_all(
+#'   vars = everything(),
+#'   func = is.numeric,
 #'   data = iris
-#' )
+#' ))
 #'
 NULL
 
 #' @export
 #' @rdname generic-expectations
-expect_allany <- function(vars,
-                          func,
-                          flt = TRUE,
-                          data = get_testdata(),
-                          args = list(),
-                          allany = c(chk_filter_all, chk_filter_any),
-                          func_desc = NULL) {
+expect_all <- function(vars,
+                       func,
+                       flt = TRUE,
+                       data = get_testdata(),
+                       args = list(),
+                       func_desc = NULL) {
+  check_expect_data_pipe(enquo(vars))
+  act <- quasi_label(enquo(data))
+  act$func_desc <- if (is.null(func_desc)) paste0("`", as_label(enquo(func)), "`") else func_desc
+  act$var_desc  <- as_label_vars(enquo(vars))
+  act$flt_desc  <- as_label_flt(enquo(flt))
+  act$args_desc <- expr_deparse_repl(args, "(^<list: |>$)", "")
+
+  act$result <- chk_filter(
+    eval_tidy(enquo(data)),
+    {{ vars }},
+    eval_tidy(enquo(func)),
+    {{ flt }},
+    args
+  )
+
+  act$var_fail <- act$result %>%
+    select_if(~any(!., na.rm=TRUE)) %>%
+    names()
+
+  act$result <- apply(act$result, 1, all)
 
+  expect_custom(
+    all(act$result, na.rm = TRUE),
+    glue("{act$lab} has {sum(!act$result, na.rm = TRUE)} records failing \\
+          {act$func_desc} on variable \\
+         {as_english_list(paste0('`', act$var_fail, '`'), 'and/or')}.
+          Vars: {act$var_desc}
```

Comment:
Actually don't like `Vars` here. Maybe `Columns` to match the docs? Or `Variable set`?

Reply:
Hm, I definitely prefer `Vars` to `Columns`. It's just presenting what's passed through as the `vars` argument, so makes sense to me to label it as such. `Variable set` seems like a reasonable middle ground though

Reply:
Yeah actually `Vars` is probably fine. Disregard!

Reply:
Ah, I didn't see this. Already changed to variable set :p Easy to change back. I'm not fussed

Reply:
Variable set it is 😉